app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+      @user = User.from_omniauth(request.env['omniauth.auth'])
+
+      if @user.persisted?
+        flash[:notice] = I18n.t 'devise.omniauth_callbacks.success', kind: 'Google'
+        sign_in_and_redirect @user, event: :authentication
+      else
+        session['devise.google_data'] = request.env['omniauth.auth'].except('extra') # Removing extra as it can overflow some session stores
+        redirect_to new_user_registration_url, alert: @user.errors.full_messages.join("\n")
+      end
+  end
+
+  def github
+    @user = User.from_omniauth(request.env['omniauth.auth'])
+
+    if @user.persisted?
+      flash[:notice] = I18n.t 'devise.omniauth_callbacks.success', kind: 'Github'
+      sign_in_and_redirect @user, event: :authentication
+    else
+      session['devise.github_data'] = request.env['omniauth.auth'].except('extra') # Removing extra as it can overflow some session stores
+      redirect_to new_user_registration_url, alert: @user.errors.full_messages.join("\n")
+    end
+  end
```

Comment:
These 2 endpoint are quite similar, I think we can DRY them a bit (with a private method?)